src/Vault.sol

```diff
@@ -910,6 +936,59 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
       _collateralAssets == 0 ? 0 : _shares.mulDiv(_collateralAssets, _depositedAssets, _rounding);
   }
 
+  /**
+   * @notice Convert Vault shares to YieldVault shares.
+   * @param _shares Vault shares to convert
+   * @param _rounding Rounding mode (i.e. down or up)
+   * @return uint256 YieldVault shares
+   */
+  function _convertToYieldVaultShares(
+    uint256 _shares,
+    Math.Rounding _rounding
+  ) internal view returns (uint256) {
+    uint256 _depositedAssets = _totalSupply();
+
+    // When the Vault is collateralized, the yield is excluded from the redeemable amount
+    uint256 _redeemableShares = _isVaultCollateralized()
+      ? _yieldVault.maxRedeem(address(this)) -
+        (_yieldVault.convertToShares(_totalAssets() - _depositedAssets))
+      : _yieldVault.maxRedeem(address(this));
```

Comment:
memoize as much as you can here.  Lots of external calls

It would be good to have an _isVaultCollateralized fxn you can pass in the results of `_totalSupply` and `_totalAssets` because you always need them in the context of the call anyway